index.js

```diff
@@ -1,100 +1,253 @@
+    getSkill(`${__dirname}/icons/${obj.skill}.png`).then(response => {
+      skills.push({'img': `data:image/png;base64,${response.toString('base64')}`, 'xp': obj.xp});
+    })
+  ));
+  return skills;
+}
+
+const getSkill = async(file)=>{
+  const result = await fs.readFile(file)
+  return result;
+}
+
+async function loadSkills(objs, ctx, offset) {
+  await Promise.all(objs.map((obj, index) =>
+    loadImage(obj.img).then((imageObj) => {
+      paintSkills(ctx, imageObj, index, offset, `${obj.xp} xp`);
+    })
+  ));
+}
+
+function paintSkills(ctx, image, index, offset, xp){
+  const yOffset = offset + 40 + (50 * Math.floor(index / 5));
+  const xOffset = 15 + (60 * (index % 5));
+  ctx.drawImage(image, xOffset, yOffset, 25, 25);
+  ctx.font = "16px Runescape";
+  ctx.textAlign = "center";
+  ctx.fillStyle = "#ffff00";
+  const xpInt = parseInt(xp.replace(/,/g, ''));
+  let xpGained = '';
+  if (xpInt >= 1000000) {
+    xpGained = `${Math.floor(xpInt*10 / 1000000)/10}m xp`;
+  }
+  else if (xpInt > 1000) {
+    xpGained = `${Math.trunc((xpInt * 10) / 1000 / 10)}k xp`;
+  } else {
+    xpGained = `${xpInt} xp`;
+  }
+  ctx.strokeText(xpGained, xOffset+15, yOffset+35);
+  ctx.fillText(xpGained, xOffset+15, yOffset+35);
+  // const start = Math.floor(Math.random() * 99);
+  // const finish = Math.floor(Math.random() * (99 - start) + start); 
+  // ctx.font = "14px Runescape";
+  // ctx.strokeText(`${start}-${finish}`, xOffset+15, yOffset+50);
+  // ctx.fillText(`${start}-${finish}`, xOffset+15, yOffset+50);
+}
+
+
+function paintIcons(ctx, image, index, count){
+  const xOffset = 15 + (40 * (index % 7));
+  const row = Math.floor(index/7);
+  const yOffset = 115 + row*35;
+  ctx.drawImage(image, xOffset, yOffset);
+  ctx.font = "14px Runescape";
+  ctx.textAlign = "right";
+  if (count >= 1000000) {
+    count = `${Math.floor(count *10 / 1000000)/ 10}m`;
+  }
+  else if (count > 100000) {
+    count = `${Math.trunc(count/ 1000)}k`;
+  }
+  else if (count > 1000) {
     count = `${Math.trunc((count * 10) / 1000) / 10}k`;
-    }
-    ctx.lineWidth = 3;
-    ctx.strokeText(count, x + offset + 30, 150);
-    ctx.fillText(count, x + offset+ 30, 150)
+  }
+  ctx.lineWidth = 2;
+  ctx.strokeText(count, xOffset+30, yOffset+30);
+  ctx.fillText(count, xOffset+30, yOffset+30)
 }
 
 async function getMultiple(objectsToGet) {
-  console.log(objectsToGet);
   let items = [];
+  objectsToGet.loot.forEach((item) => {
+    const coins = [617,995,996,997,998,999,1000,1001,1002,1003,1004,6964,8890,8891,8892,8893,8894,8895,8896,8897,8898,8899,14440,18028];
+    if(coins.indexOf(item.id) != -1){
+      if (item.count > 50000){
+        item.id = 1004;
+      } else if (item.count > 10000){
+        item.id = 1003;
+      } else if (item.count > 1000 ){
+        item.id = 1001;
+      } else {
+        item.id = 998;
+      }
+    }
+  });
   await Promise.all(objectsToGet.loot.map(obj =>
     axios.get(`https://api.osrsbox.com/items/${obj.id}`).then(response => {
```

Comment:
IMHO host and path for apis should be extracted to globals in case they change later.